File: presentation/src/main/java/by/grodno/vasili/githubsimpleapp/feature/user_details/UserDetailsItem.java
package by.grodno.vasili.githubsimpleapp.feature.user_details;

import java.util.List;

/**
 * User model for representing in details activity
 */
public class UserDetailsItem {
    public String id;
    public String login;
    public String name;
    public String avatarUrl;
    public String email;
    public List<OrganizationItem> organizations;
    public Integer followingCount;
    public Integer followersCount;
    public String created;

    UserDetailsItem(String id, String login, String name, String avatarUrl, String email, List<OrganizationItem> organizations, Integer followingCount, Integer followersCount, String created) {
        this.id = id;
        this.login = login;
        this.name = name;
        this.avatarUrl = avatarUrl;
        this.email = email;
        this.organizations = organizations;
        this.followingCount = followingCount;
        this.followersCount = followersCount;
        this.created = created;
    }
}

File: presentation/src/main/java/by/grodno/vasili/githubsimpleapp/feature/user_details/UserDetailsItem.kt
package by.grodno.vasili.githubsimpleapp.feature.user_details

import by.grodno.vasili.domain.model.User

/**
 * User model for representing in details activity.
 */
data class UserDetailsItem(
        val id: String,
        val login: String,
        val name: String,
        val avatarUrl: String,
        val email: String,
        val organizations: List<OrganizationItem>,
        val followingCount: Int,
        val followersCount: Int,
        val created: String
) {
    constructor(user: User) : this(
            id = user.id,
            login = user.login,
            name = user.name,
            avatarUrl = user.avatarUrl,
            email = user.email,
            organizations = user.organizations.map { OrganizationItem(it) },
            followingCount = user.followingCount,
            followersCount = user.followersCount,
            created = user.created
    )
}
